pkg/activator/handler/main_test.go

```diff
@@ -114,3 +120,152 @@ func BenchmarkHandlerChain(b *testing.B) {
 		})
 	})
 }
+
+func TestProxyWithChainHandler(t *testing.T) {
```

Comment:
This is the issue reproducer utilizing the activator's handler.

Reply:
Reading this test in isolation it's not clear what it is doing - can you add a detailed comment and maybe link out to the knative and golang issue?